src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                                                     GetHttpContext(),
+                                                     exposedColumnNames,
+                                                     dataSourceName);
+
+            dbResultSetRow = dbResultSet is not null ? (dbResultSet.Rows.FirstOrDefault() ?? new DbResultSetRow()) : null;
+
+            if (dbResultSetRow is null || dbResultSetRow.Columns.Count == 0)
+            {
+                if (isLinkingEntity)
+                {
+                    throw new DataApiBuilderException(message: $"Could not insert row with given values in the linking table",
+                                                      statusCode: HttpStatusCode.InternalServerError,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed);
+                }
+                else
+                {
+                    if (dbResultSetRow is null)
+                    {
+                        throw new DataApiBuilderException(message: "No data returned back from database.",
+                                                          statusCode: HttpStatusCode.InternalServerError,
+                                                          subStatusCode: DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed);
+                    }
+                    else
+                    {
+                        throw new DataApiBuilderException(message: $"Could not insert row with given values for entity: {entityName}",
+                                                          statusCode: HttpStatusCode.Forbidden,
+                                                          subStatusCode: DataApiBuilderException.SubStatusCodes.DatabasePolicyFailure);
+                    }
+                }
+            }
+
+            return dbResultSetRow.Columns;
+        }
+
+        /// <summary>
+        /// Identifies and returns the foreign key definition with the right combination of
+        /// referencing and referenced entity names.
+        /// </summary>
+        /// <param name="sqlMetadataProvider">SqlMetadaProvider object for the given database</param>
+        /// <param name="foreignKeyDefinitions">List of foreign key definitions</param>
+        /// <param name="referencingEntityName">Referencing entity name</param>
+        /// <param name="referencedEntityName">Referenced entity name</param>
+        /// <param name="isMToNRelationship">Indicates whether the relationship is M:N</param>
+        /// <returns>Valid forrign key definition with the right referencing and referencing entities</returns>
+        /// <exception cref="DataApiBuilderException"></exception>
+        private static ForeignKeyDefinition FetchValidForeignKeyDefinition(ISqlMetadataProvider sqlMetadataProvider, List<ForeignKeyDefinition> foreignKeyDefinitions, string referencingEntityName, string referencedEntityName, bool isMToNRelationship)
```

Comment:
This method seems more appropriate to be placed in MetadataProvider. In the request validation PR, I have added a method `TryGetFKDefinition()` in the `ISqlMetadatProvider` class which does this exact same thing. But currently does not take care of M:N relationship. Maybe you can reuse/build on top of it?

Reply:
Added `TryGetFKDefinition()` in `ISqlMetadaProvider`.  Added logic to handle M:N relationship type.